great_expectations/rule_based_profiler/parameter_builder/histogram_parameter_builder.py

```diff
@@ -217,11 +227,38 @@ def _build_parameters(
         weights: list
 
         if is_categorical:
-            fully_qualified_column_value_counts_metric_single_batch_parameter_builder_name: str = f"{RAW_PARAMETER_KEY}{self._column_value_counts_metric_single_batch_parameter_builder_config.name}"
+            """
+            In this scenario, even though initial intent was to compute histogram, data turned out to be categorical.
```

Comment:
@jcampbell Yes, it is indeed there in the constructor if  `bucketize_data` was `False` initially.  But it would not be declared if `bucketize_data` were `True` initially.  This is why here, since we determined that the `bins` are flawed and have to make `bucketize_data` `False` (i.e., `is_categorical` became `True`), we need to instantiate and execute this `"column.value_counts"` parameter builder (because it was not created initially).  Thanks.

Reply:
Sounds like this is the same as the below question then wrt what the case is that we're trying to handle.

Reply:
> Sounds like this is the same as the below question then wrt what the case is that we're trying to handle.

@jcampbell Yes, it is -- but because we decided to declare metric computations "parsimoniously", we do not declare them, until we are absolutely sure that we will need for them to be computed.  (In the old implementation, we computed both, irrespective of the final value of `is_categorical`).